Build fails on Debian for architecture armel, amrhf, mips & mipsel with "error: template-id 'ink_atomic_swap<int64_t>' used as a declarator" error

All theses architecture fail with same error:
```Making all in records
make[3]: Entering directory '/<<PKGBUILDDIR>>/lib/records'
c++ -DHAVE_CONFIG_H -I. -I../../include  -Dlinux -D_LARGEFILE64_SOURCE=1 -D_FILE_OFFSET_BITS=64 -D_REENTRANT -D__STDC_LIMIT_MACROS=1 -D__STDC_FORMAT_MACROS=1 -I/<<PKGBUILDDIR>>/iocore/eventsystem -I/<<PKGBUILDDIR>>/iocore/utils -I/<<PKGBUILDDIR>>/mgmt -I/<<PKGBUILDDIR>>/mgmt/api/include -I/<<PKGBUILDDIR>>/mgmt/utils -I/<<PKGBUILDDIR>>/include -I/<<PKGBUILDDIR>>/lib -I/usr/include/tcl8.6 -Wdate-time -D_FORTIFY_SOURCE=2 -D_GNU_SOURCE -DOPENSSL_NO_SSL_INTERN -std=c++17 -g -pipe -Wall -Wextra -Wno-ignored-qualifiers -Wno-unused-parameter -Wno-format-truncation -Wno-cast-function-type -Wno-stringop-overflow -feliminate-unused-debug-symbols -fno-strict-aliasing -Wno-invalid-offsetof -Wno-noexcept-type -g -O2 -fdebug-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security -O3 -c -o P_RecCore.o P_RecCore.cc
c++ -DHAVE_CONFIG_H -I. -I../../include  -Dlinux -D_LARGEFILE64_SOURCE=1 -D_FILE_OFFSET_BITS=64 -D_REENTRANT -D__STDC_LIMIT_MACROS=1 -D__STDC_FORMAT_MACROS=1 -I/<<PKGBUILDDIR>>/iocore/eventsystem -I/<<PKGBUILDDIR>>/iocore/utils -I/<<PKGBUILDDIR>>/mgmt -I/<<PKGBUILDDIR>>/mgmt/api/include -I/<<PKGBUILDDIR>>/mgmt/utils -I/<<PKGBUILDDIR>>/include -I/<<PKGBUILDDIR>>/lib -I/usr/include/tcl8.6 -Wdate-time -D_FORTIFY_SOURCE=2 -D_GNU_SOURCE -DOPENSSL_NO_SSL_INTERN -std=c++17 -g -pipe -Wall -Wextra -Wno-ignored-qualifiers -Wno-unused-parameter -Wno-format-truncation -Wno-cast-function-type -Wno-stringop-overflow -feliminate-unused-debug-symbols -fno-strict-aliasing -Wno-invalid-offsetof -Wno-noexcept-type -g -O2 -fdebug-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security -O3 -c -o RecConfigParse.o RecConfigParse.cc
c++ -DHAVE_CONFIG_H -I. -I../../include  -Dlinux -D_LARGEFILE64_SOURCE=1 -D_FILE_OFFSET_BITS=64 -D_REENTRANT -D__STDC_LIMIT_MACROS=1 -D__STDC_FORMAT_MACROS=1 -I/<<PKGBUILDDIR>>/iocore/eventsystem -I/<<PKGBUILDDIR>>/iocore/utils -I/<<PKGBUILDDIR>>/mgmt -I/<<PKGBUILDDIR>>/mgmt/api/include -I/<<PKGBUILDDIR>>/mgmt/utils -I/<<PKGBUILDDIR>>/include -I/<<PKGBUILDDIR>>/lib -I/usr/include/tcl8.6 -Wdate-time -D_FORTIFY_SOURCE=2 -D_GNU_SOURCE -DOPENSSL_NO_SSL_INTERN -std=c++17 -g -pipe -Wall -Wextra -Wno-ignored-qualifiers -Wno-unused-parameter -Wno-format-truncation -Wno-cast-function-type -Wno-stringop-overflow -feliminate-unused-debug-symbols -fno-strict-aliasing -Wno-invalid-offsetof -Wno-noexcept-type -g -O2 -fdebug-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security -O3 -c -o RecCore.o RecCore.cc
c++ -DHAVE_CONFIG_H -I. -I../../include  -Dlinux -D_LARGEFILE64_SOURCE=1 -D_FILE_OFFSET_BITS=64 -D_REENTRANT -D__STDC_LIMIT_MACROS=1 -D__STDC_FORMAT_MACROS=1 -I/<<PKGBUILDDIR>>/iocore/eventsystem -I/<<PKGBUILDDIR>>/iocore/utils -I/<<PKGBUILDDIR>>/mgmt -I/<<PKGBUILDDIR>>/mgmt/api/include -I/<<PKGBUILDDIR>>/mgmt/utils -I/<<PKGBUILDDIR>>/include -I/<<PKGBUILDDIR>>/lib -I/usr/include/tcl8.6 -Wdate-time -D_FORTIFY_SOURCE=2 -D_GNU_SOURCE -DOPENSSL_NO_SSL_INTERN -std=c++17 -g -pipe -Wall -Wextra -Wno-ignored-qualifiers -Wno-unused-parameter -Wno-format-truncation -Wno-cast-function-type -Wno-stringop-overflow -feliminate-unused-debug-symbols -fno-strict-aliasing -Wno-invalid-offsetof -Wno-noexcept-type -g -O2 -fdebug-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security -O3 -c -o RecDebug.o RecDebug.cc
In file included from P_RecUtils.h:27,
                 from RecDebug.cc:24:
../../include/tscore/ink_atomic.h:96:26: error: template-id 'ink_atomic_swap<int64_t>' used as a declarator
 ink_atomic_swap<int64_t>(pvint64 mem, int64_t value)
                          ^~~~~~~
../../include/tscore/ink_atomic.h:96:26: error: 'pvint64' was not declared in this scope
In file included from P_RecUtils.h:27,
                 from P_RecCore.cc:33:
../../include/tscore/ink_atomic.h:96:26: error: template-id 'ink_atomic_swap<int64_t>' used as a declarator
 ink_atomic_swap<int64_t>(pvint64 mem, int64_t value)
                          ^~~~~~~
../../include/tscore/ink_atomic.h:96:26: note: suggested alternative: 'printf'
 ink_atomic_swap<int64_t>(pvint64 mem, int64_t value)
                          ^~~~~~~
                          printf
../../include/tscore/ink_atomic.h:96:47: error: expected primary-expression before 'value'
 ink_atomic_swap<int64_t>(pvint64 mem, int64_t value)
                                               ^~~~~
../../include/tscore/ink_atomic.h:96:52: error: expression list treated as compound expression in initializer [-fpermissive]
 ink_atomic_swap<int64_t>(pvint64 mem, int64_t value)
                                                    ^
../../include/tscore/ink_atomic.h:96:1: warning: too many template headers for 'ink_atomic_swap' (should be 0)
 ink_atomic_swap<int64_t>(pvint64 mem, int64_t value)
 ^~~~~~~~~~~~~~~~~~~~~~~~
../../include/tscore/ink_atomic.h:96:53: error: expected ';' before '{' token
 ink_atomic_swap<int64_t>(pvint64 mem, int64_t value)
                                                     ^
                                                     ;
 {
 ~                                                    ```

To be fully acurate, here's the full list of errors:
```../../include/tscore/ink_atomic.h:96:26: error: template-id 'ink_atomic_swap<int64_t>' used as a declarator
 ink_atomic_swap<int64_t>(pvint64 mem, int64_t value)
                          ^~~~~~~
In file included from P_RecUtils.h:27,
                 from RecConfigParse.cc:34:
../../include/tscore/ink_atomic.h:96:26: error: template-id 'ink_atomic_swap<int64_t>' used as a declarator
 ink_atomic_swap<int64_t>(pvint64 mem, int64_t value)
                          ^~~~~~~
../../include/tscore/ink_atomic.h:96:26: error: 'pvint64' was not declared in this scope
../../include/tscore/ink_atomic.h:96:26: error: 'pvint64' was not declared in this scope
../../include/tscore/ink_atomic.h:96:26: note: suggested alternative: 'printf'
 ink_atomic_swap<int64_t>(pvint64 mem, int64_t value)
                          ^~~~~~~
                          printf
../../include/tscore/ink_atomic.h:96:47: error: expected primary-expression before 'value'
 ink_atomic_swap<int64_t>(pvint64 mem, int64_t value)
                                               ^~~~~
../../include/tscore/ink_atomic.h:96:52: error: expression list treated as compound expression in initializer [-fpermissive]
 ink_atomic_swap<int64_t>(pvint64 mem, int64_t value)
                                                    ^
../../include/tscore/ink_atomic.h:96:1: warning: too many template headers for 'ink_atomic_swap' (should be 0)
 ink_atomic_swap<int64_t>(pvint64 mem, int64_t value)
 ^~~~~~~~~~~~~~~~~~~~~~~~
../../include/tscore/ink_atomic.h:96:53: error: expected ';' before '{' token
 ink_atomic_swap<int64_t>(pvint64 mem, int64_t value)
                                                     ^
                                                     ;
 {
 ~                                                    
../../include/tscore/ink_atomic.h:108:25: error: template-id 'ink_atomic_cas<int64_t>' used as a declarator
 ink_atomic_cas<int64_t>(pvint64 mem, int64_t old, int64_t new_value)
                         ^~~~~~~
../../include/tscore/ink_atomic.h:108:25: error: 'pvint64' was not declared in this scope
../../include/tscore/ink_atomic.h:96:26: note: suggested alternative: 'printf'
 ink_atomic_swap<int64_t>(pvint64 mem, int64_t value)
                          ^~~~~~~
                          printf
../../include/tscore/ink_atomic.h:96:47: error: expected primary-expression before 'value'
 ink_atomic_swap<int64_t>(pvint64 mem, int64_t value)
                                               ^~~~~
../../include/tscore/ink_atomic.h:96:52: error: expression list treated as compound expression in initializer [-fpermissive]
 ink_atomic_swap<int64_t>(pvint64 mem, int64_t value)
                                                    ^
../../include/tscore/ink_atomic.h:96:1: warning: too many template headers for 'ink_atomic_swap' (should be 0)
 ink_atomic_swap<int64_t>(pvint64 mem, int64_t value)
 ^~~~~~~~~~~~~~~~~~~~~~~~
../../include/tscore/ink_atomic.h:96:53: error: expected ';' before '{' token
 ink_atomic_swap<int64_t>(pvint64 mem, int64_t value)
                                                     ^
                                                     ;
 {
 ~                                                    
../../include/tscore/ink_atomic.h:108:25: error: template-id 'ink_atomic_cas<int64_t>' used as a declarator
 ink_atomic_cas<int64_t>(pvint64 mem, int64_t old, int64_t new_value)
                         ^~~~~~~
../../include/tscore/ink_atomic.h:108:25: error: 'pvint64' was not declared in this scope
../../include/tscore/ink_atomic.h:108:25: note: suggested alternative: 'printf'
 ink_atomic_cas<int64_t>(pvint64 mem, int64_t old, int64_t new_value)
                         ^~~~~~~
                         printf
../../include/tscore/ink_atomic.h:108:46: error: expected primary-expression before 'old'
 ink_atomic_cas<int64_t>(pvint64 mem, int64_t old, int64_t new_value)
                                              ^~~
../../include/tscore/ink_atomic.h:108:59: error: expected primary-expression before 'new_value'
 ink_atomic_cas<int64_t>(pvint64 mem, int64_t old, int64_t new_value)
                                                           ^~~~~~~~~
../../include/tscore/ink_atomic.h:108:68: error: expression list treated as compound expression in initializer [-fpermissive]
 ink_atomic_cas<int64_t>(pvint64 mem, int64_t old, int64_t new_value)
                                                                    ^
../../include/tscore/ink_atomic.h:108:1: warning: too many template headers for 'ink_atomic_cas' (should be 0)
 ink_atomic_cas<int64_t>(pvint64 mem, int64_t old, int64_t new_value)
 ^~~~~~~~~~~~~~~~~~~~~~~
../../include/tscore/ink_atomic.h:108:69: error: expected ';' before '{' token
 ink_atomic_cas<int64_t>(pvint64 mem, int64_t old, int64_t new_value)
                                                                     ^
                                                                     ;
 {
 ~                                                                    
../../include/tscore/ink_atomic.h:123:22: error: 'template<class Amount> int64_t ink_atomic_increment' conflicts with a previous declaration
 ink_atomic_increment(pvint64 mem, Amount value)
                      ^~~~~~~
../../include/tscore/ink_atomic.h:76:1: note: previous declaration 'Type ink_atomic_increment(Type*, Amount)'
 ink_atomic_increment(Type *mem, Amount count)
 ^~~~~~~~~~~~~~~~~~~~
../../include/tscore/ink_atomic.h:123:22: error: 'pvint64' was not declared in this scope
 ink_atomic_increment(pvint64 mem, Amount value)
                      ^~~~~~~
../../include/tscore/ink_atomic.h:108:25: note: suggested alternative: 'printf'
 ink_atomic_cas<int64_t>(pvint64 mem, int64_t old, int64_t new_value)
                         ^~~~~~~
                         printf
../../include/tscore/ink_atomic.h:108:46: error: expected primary-expression before 'old'
 ink_atomic_cas<int64_t>(pvint64 mem, int64_t old, int64_t new_value)
                                              ^~~
../../include/tscore/ink_atomic.h:108:59: error: expected primary-expression before 'new_value'
 ink_atomic_cas<int64_t>(pvint64 mem, int64_t old, int64_t new_value)
                                                           ^~~~~~~~~
../../include/tscore/ink_atomic.h:108:68: error: expression list treated as compound expression in initializer [-fpermissive]
 ink_atomic_cas<int64_t>(pvint64 mem, int64_t old, int64_t new_value)
                                                                    ^
../../include/tscore/ink_atomic.h:108:1: warning: too many template headers for 'ink_atomic_cas' (should be 0)
 ink_atomic_cas<int64_t>(pvint64 mem, int64_t old, int64_t new_value)
 ^~~~~~~~~~~~~~~~~~~~~~~
../../include/tscore/ink_atomic.h:108:69: error: expected ';' before '{' token
 ink_atomic_cas<int64_t>(pvint64 mem, int64_t old, int64_t new_value)
                                                                     ^
                                                                     ;
 {
 ~                                                                    
../../include/tscore/ink_atomic.h:123:22: error: 'template<class Amount> int64_t ink_atomic_increment' conflicts with a previous declaration
 ink_atomic_increment(pvint64 mem, Amount value)
                      ^~~~~~~
../../include/tscore/ink_atomic.h:76:1: note: previous declaration 'Type ink_atomic_increment(Type*, Amount)'
 ink_atomic_increment(Type *mem, Amount count)
 ^~~~~~~~~~~~~~~~~~~~
../../include/tscore/ink_atomic.h:123:22: error: 'pvint64' was not declared in this scope
 ink_atomic_increment(pvint64 mem, Amount value)
                      ^~~~~~~
In file included from P_RecUtils.h:27,
                 from RecDebug.cc:24:
../../include/tscore/ink_atomic.h:96:26: error: template-id 'ink_atomic_swap<int64_t>' used as a declarator
 ink_atomic_swap<int64_t>(pvint64 mem, int64_t value)
                          ^~~~~~~
../../include/tscore/ink_atomic.h:96:26: error: 'pvint64' was not declared in this scope
../../include/tscore/ink_atomic.h:123:22: note: suggested alternative: 'printf'
 ink_atomic_increment(pvint64 mem, Amount value)
                      ^~~~~~~
                      printf
../../include/tscore/ink_atomic.h:123:42: error: expected primary-expression before 'value'
 ink_atomic_increment(pvint64 mem, Amount value)
                                          ^~~~~
../../include/tscore/ink_atomic.h:123:47: error: expression list treated as compound expression in initializer [-fpermissive]
 ink_atomic_increment(pvint64 mem, Amount value)
                                               ^
../../include/tscore/ink_atomic.h:123:48: error: expected ';' before '{' token
 ink_atomic_increment(pvint64 mem, Amount value)
                                                ^
                                                ;
 {
 ~                                               
../../include/tscore/ink_atomic.h:135:22: error: 'template<class Amount> int64_t ink_atomic_decrement' conflicts with a previous declaration
 ink_atomic_decrement(pvint64 mem, Amount value)
                      ^~~~~~~
../../include/tscore/ink_atomic.h:85:1: note: previous declaration 'Type ink_atomic_decrement(Type*, Amount)'
 ink_atomic_decrement(Type *mem, Amount count)
 ^~~~~~~~~~~~~~~~~~~~
../../include/tscore/ink_atomic.h:135:22: error: 'pvint64' was not declared in this scope
 ink_atomic_decrement(pvint64 mem, Amount value)
                      ^~~~~~~
../../include/tscore/ink_atomic.h:123:22: note: suggested alternative: 'printf'
 ink_atomic_increment(pvint64 mem, Amount value)
                      ^~~~~~~
                      printf
../../include/tscore/ink_atomic.h:123:42: error: expected primary-expression before 'value'
 ink_atomic_increment(pvint64 mem, Amount value)
                                          ^~~~~
../../include/tscore/ink_atomic.h:123:47: error: expression list treated as compound expression in initializer [-fpermissive]
 ink_atomic_increment(pvint64 mem, Amount value)
                                               ^
../../include/tscore/ink_atomic.h:96:26: note: suggested alternative: 'printf'
 ink_atomic_swap<int64_t>(pvint64 mem, int64_t value)
                          ^~~~~~~
                          printf
../../include/tscore/ink_atomic.h:96:47: error: expected primary-expression before 'value'
 ink_atomic_swap<int64_t>(pvint64 mem, int64_t value)
                                               ^~~~~
../../include/tscore/ink_atomic.h:96:52: error: expression list treated as compound expression in initializer [-fpermissive]
 ink_atomic_swap<int64_t>(pvint64 mem, int64_t value)
                                                    ^
../../include/tscore/ink_atomic.h:96:1: warning: too many template headers for 'ink_atomic_swap' (should be 0)
 ink_atomic_swap<int64_t>(pvint64 mem, int64_t value)
 ^~~~~~~~~~~~~~~~~~~~~~~~
../../include/tscore/ink_atomic.h:96:53: error: expected ';' before '{' token
 ink_atomic_swap<int64_t>(pvint64 mem, int64_t value)
                                                     ^
                                                     ;
 {
 ~                                                    
../../include/tscore/ink_atomic.h:108:25: error: template-id 'ink_atomic_cas<int64_t>' used as a declarator
 ink_atomic_cas<int64_t>(pvint64 mem, int64_t old, int64_t new_value)
                         ^~~~~~~
../../include/tscore/ink_atomic.h:108:25: error: 'pvint64' was not declared in this scope
../../include/tscore/ink_atomic.h:123:48: error: expected ';' before '{' token
 ink_atomic_increment(pvint64 mem, Amount value)
                                                ^
                                                ;
 {
 ~                                               
../../include/tscore/ink_atomic.h:135:22: error: 'template<class Amount> int64_t ink_atomic_decrement' conflicts with a previous declaration
 ink_atomic_decrement(pvint64 mem, Amount value)
                      ^~~~~~~
../../include/tscore/ink_atomic.h:85:1: note: previous declaration 'Type ink_atomic_decrement(Type*, Amount)'
 ink_atomic_decrement(Type *mem, Amount count)
 ^~~~~~~~~~~~~~~~~~~~
../../include/tscore/ink_atomic.h:135:22: error: 'pvint64' was not declared in this scope
 ink_atomic_decrement(pvint64 mem, Amount value)
                      ^~~~~~~
../../include/tscore/ink_atomic.h:135:22: note: suggested alternative: 'printf'
 ink_atomic_decrement(pvint64 mem, Amount value)
                      ^~~~~~~
                      printf
../../include/tscore/ink_atomic.h:135:42: error: expected primary-expression before 'value'
 ink_atomic_decrement(pvint64 mem, Amount value)
                                          ^~~~~
../../include/tscore/ink_atomic.h:135:47: error: expression list treated as compound expression in initializer [-fpermissive]
 ink_atomic_decrement(pvint64 mem, Amount value)
                                               ^
../../include/tscore/ink_atomic.h:135:48: error: expected ';' before '{' token
 ink_atomic_decrement(pvint64 mem, Amount value)
                                                ^
                                                ;
 {
 ~                                               
../../include/tscore/ink_atomic.h:108:25: note: suggested alternative: 'printf'
 ink_atomic_cas<int64_t>(pvint64 mem, int64_t old, int64_t new_value)
                         ^~~~~~~
                         printf
../../include/tscore/ink_atomic.h:147:22: error: 'template<class Amount> uint64_t ink_atomic_increment' conflicts with a previous declaration
 ink_atomic_increment(pvuint64 mem, Amount value)
                      ^~~~~~~~
../../include/tscore/ink_atomic.h:76:1: note: previous declaration 'Type ink_atomic_increment(Type*, Amount)'
 ink_atomic_increment(Type *mem, Amount count)
 ^~~~~~~~~~~~~~~~~~~~
../../include/tscore/ink_atomic.h:147:22: error: 'pvuint64' was not declared in this scope
 ink_atomic_increment(pvuint64 mem, Amount value)
                      ^~~~~~~~
../../include/tscore/ink_atomic.h:108:46: error: expected primary-expression before 'old'
 ink_atomic_cas<int64_t>(pvint64 mem, int64_t old, int64_t new_value)
                                              ^~~
../../include/tscore/ink_atomic.h:108:59: error: expected primary-expression before 'new_value'
 ink_atomic_cas<int64_t>(pvint64 mem, int64_t old, int64_t new_value)
                                                           ^~~~~~~~~
../../include/tscore/ink_atomic.h:108:68: error: expression list treated as compound expression in initializer [-fpermissive]
 ink_atomic_cas<int64_t>(pvint64 mem, int64_t old, int64_t new_value)
                                                                    ^
../../include/tscore/ink_atomic.h:108:1: warning: too many template headers for 'ink_atomic_cas' (should be 0)
 ink_atomic_cas<int64_t>(pvint64 mem, int64_t old, int64_t new_value)
 ^~~~~~~~~~~~~~~~~~~~~~~
../../include/tscore/ink_atomic.h:135:22: note: suggested alternative: 'printf'
 ink_atomic_decrement(pvint64 mem, Amount value)
                      ^~~~~~~
                      printf
../../include/tscore/ink_atomic.h:108:69: error: expected ';' before '{' token
 ink_atomic_cas<int64_t>(pvint64 mem, int64_t old, int64_t new_value)
                                                                     ^
                                                                     ;
 {
 ~                                                                    
../../include/tscore/ink_atomic.h:123:22: error: 'template<class Amount> int64_t ink_atomic_increment' conflicts with a previous declaration
 ink_atomic_increment(pvint64 mem, Amount value)
                      ^~~~~~~
../../include/tscore/ink_atomic.h:76:1: note: previous declaration 'Type ink_atomic_increment(Type*, Amount)'
 ink_atomic_increment(Type *mem, Amount count)
 ^~~~~~~~~~~~~~~~~~~~
../../include/tscore/ink_atomic.h:123:22: error: 'pvint64' was not declared in this scope
 ink_atomic_increment(pvint64 mem, Amount value)
                      ^~~~~~~
../../include/tscore/ink_atomic.h:135:42: error: expected primary-expression before 'value'
 ink_atomic_decrement(pvint64 mem, Amount value)
                                          ^~~~~
../../include/tscore/ink_atomic.h:135:47: error: expression list treated as compound expression in initializer [-fpermissive]
 ink_atomic_decrement(pvint64 mem, Amount value)
                                               ^
../../include/tscore/ink_atomic.h:135:48: error: expected ';' before '{' token
 ink_atomic_decrement(pvint64 mem, Amount value)
                                                ^
                                                ;
 {
 ~                                               
../../include/tscore/ink_atomic.h:147:22: error: 'template<class Amount> uint64_t ink_atomic_increment' conflicts with a previous declaration
 ink_atomic_increment(pvuint64 mem, Amount value)
                      ^~~~~~~~
../../include/tscore/ink_atomic.h:76:1: note: previous declaration 'Type ink_atomic_increment(Type*, Amount)'
 ink_atomic_increment(Type *mem, Amount count)
 ^~~~~~~~~~~~~~~~~~~~
../../include/tscore/ink_atomic.h:147:22: error: 'pvuint64' was not declared in this scope
 ink_atomic_increment(pvuint64 mem, Amount value)
                      ^~~~~~~~
../../include/tscore/ink_atomic.h:123:22: note: suggested alternative: 'printf'
 ink_atomic_increment(pvint64 mem, Amount value)
                      ^~~~~~~
                      printf
../../include/tscore/ink_atomic.h:123:42: error: expected primary-expression before 'value'
 ink_atomic_increment(pvint64 mem, Amount value)
                                          ^~~~~
../../include/tscore/ink_atomic.h:123:47: error: expression list treated as compound expression in initializer [-fpermissive]
 ink_atomic_increment(pvint64 mem, Amount value)
                                               ^
../../include/tscore/ink_atomic.h:123:48: error: expected ';' before '{' token
 ink_atomic_increment(pvint64 mem, Amount value)
                                                ^
                                                ;
 {
 ~                                               
../../include/tscore/ink_atomic.h:147:22: note: suggested alternative: 'uint64_t'
 ink_atomic_increment(pvuint64 mem, Amount value)
                      ^~~~~~~~
                      uint64_t
../../include/tscore/ink_atomic.h:147:43: error: expected primary-expression before 'value'
 ink_atomic_increment(pvuint64 mem, Amount value)
                                           ^~~~~
../../include/tscore/ink_atomic.h:147:48: error: expression list treated as compound expression in initializer [-fpermissive]
 ink_atomic_increment(pvuint64 mem, Amount value)
                                                ^
../../include/tscore/ink_atomic.h:147:49: error: expected ';' before '{' token
 ink_atomic_increment(pvuint64 mem, Amount value)
                                                 ^
                                                 ;
 {
 ~                                                
../../include/tscore/ink_atomic.h:159:22: error: 'template<class Amount> uint64_t ink_atomic_decrement' conflicts with a previous declaration
 ink_atomic_decrement(pvuint64 mem, Amount value)
                      ^~~~~~~~
../../include/tscore/ink_atomic.h:85:1: note: previous declaration 'Type ink_atomic_decrement(Type*, Amount)'
 ink_atomic_decrement(Type *mem, Amount count)
 ^~~~~~~~~~~~~~~~~~~~
../../include/tscore/ink_atomic.h:159:22: error: 'pvuint64' was not declared in this scope
 ink_atomic_decrement(pvuint64 mem, Amount value)
                      ^~~~~~~~
../../include/tscore/ink_atomic.h:135:22: error: 'template<class Amount> int64_t ink_atomic_decrement' conflicts with a previous declaration
 ink_atomic_decrement(pvint64 mem, Amount value)
                      ^~~~~~~
../../include/tscore/ink_atomic.h:85:1: note: previous declaration 'Type ink_atomic_decrement(Type*, Amount)'
 ink_atomic_decrement(Type *mem, Amount count)
 ^~~~~~~~~~~~~~~~~~~~
../../include/tscore/ink_atomic.h:135:22: error: 'pvint64' was not declared in this scope
 ink_atomic_decrement(pvint64 mem, Amount value)
                      ^~~~~~~
../../include/tscore/ink_atomic.h:147:22: note: suggested alternative: 'uint64_t'
 ink_atomic_increment(pvuint64 mem, Amount value)
                      ^~~~~~~~
                      uint64_t
../../include/tscore/ink_atomic.h:147:43: error: expected primary-expression before 'value'
 ink_atomic_increment(pvuint64 mem, Amount value)
                                           ^~~~~
../../include/tscore/ink_atomic.h:147:48: error: expression list treated as compound expression in initializer [-fpermissive]
 ink_atomic_increment(pvuint64 mem, Amount value)
                                                ^
../../include/tscore/ink_atomic.h:147:49: error: expected ';' before '{' token
 ink_atomic_increment(pvuint64 mem, Amount value)
                                                 ^
                                                 ;
 {
 ~                                                
../../include/tscore/ink_atomic.h:159:22: error: 'template<class Amount> uint64_t ink_atomic_decrement' conflicts with a previous declaration
 ink_atomic_decrement(pvuint64 mem, Amount value)
                      ^~~~~~~~
../../include/tscore/ink_atomic.h:85:1: note: previous declaration 'Type ink_atomic_decrement(Type*, Amount)'
 ink_atomic_decrement(Type *mem, Amount count)
 ^~~~~~~~~~~~~~~~~~~~
../../include/tscore/ink_atomic.h:159:22: error: 'pvuint64' was not declared in this scope
 ink_atomic_decrement(pvuint64 mem, Amount value)
                      ^~~~~~~~
../../include/tscore/ink_atomic.h:135:22: note: suggested alternative: 'printf'
 ink_atomic_decrement(pvint64 mem, Amount value)
                      ^~~~~~~
                      printf
../../include/tscore/ink_atomic.h:135:42: error: expected primary-expression before 'value'
 ink_atomic_decrement(pvint64 mem, Amount value)
                                          ^~~~~
../../include/tscore/ink_atomic.h:135:47: error: expression list treated as compound expression in initializer [-fpermissive]
 ink_atomic_decrement(pvint64 mem, Amount value)
                                               ^
../../include/tscore/ink_atomic.h:135:48: error: expected ';' before '{' token
 ink_atomic_decrement(pvint64 mem, Amount value)
                                                ^
                                                ;
 {
 ~                                               
../../include/tscore/ink_atomic.h:147:22: error: 'template<class Amount> uint64_t ink_atomic_increment' conflicts with a previous declaration
 ink_atomic_increment(pvuint64 mem, Amount value)
                      ^~~~~~~~
../../include/tscore/ink_atomic.h:76:1: note: previous declaration 'Type ink_atomic_increment(Type*, Amount)'
 ink_atomic_increment(Type *mem, Amount count)
 ^~~~~~~~~~~~~~~~~~~~
../../include/tscore/ink_atomic.h:159:22: note: suggested alternative: 'uint64_t'
 ink_atomic_decrement(pvuint64 mem, Amount value)
                      ^~~~~~~~
                      uint64_t
../../include/tscore/ink_atomic.h:159:43: error: expected primary-expression before 'value'
 ink_atomic_decrement(pvuint64 mem, Amount value)
                                           ^~~~~
../../include/tscore/ink_atomic.h:159:48: error: expression list treated as compound expression in initializer [-fpermissive]
 ink_atomic_decrement(pvuint64 mem, Amount value)
                                                ^
../../include/tscore/ink_atomic.h:159:49: error: expected ';' before '{' token
 ink_atomic_decrement(pvuint64 mem, Amount value)
                                                 ^
                                                 ;
 {
 ~                                                
../../include/tscore/ink_atomic.h:147:22: error: 'pvuint64' was not declared in this scope
 ink_atomic_increment(pvuint64 mem, Amount value)
                      ^~~~~~~~
../../include/tscore/ink_atomic.h:159:22: note: suggested alternative: 'uint64_t'
 ink_atomic_decrement(pvuint64 mem, Amount value)
                      ^~~~~~~~
                      uint64_t
../../include/tscore/ink_atomic.h:159:43: error: expected primary-expression before 'value'
 ink_atomic_decrement(pvuint64 mem, Amount value)
                                           ^~~~~
../../include/tscore/ink_atomic.h:159:48: error: expression list treated as compound expression in initializer [-fpermissive]
 ink_atomic_decrement(pvuint64 mem, Amount value)
                                                ^
../../include/tscore/ink_atomic.h:159:49: error: expected ';' before '{' token
 ink_atomic_decrement(pvuint64 mem, Amount value)
                                                 ^
                                                 ;
 {
 ~                                                
../../include/tscore/ink_atomic.h:147:22: note: suggested alternative: 'dirent64'
 ink_atomic_increment(pvuint64 mem, Amount value)
                      ^~~~~~~~
                      dirent64
../../include/tscore/ink_atomic.h:147:43: error: expected primary-expression before 'value'
 ink_atomic_increment(pvuint64 mem, Amount value)
                                           ^~~~~
../../include/tscore/ink_atomic.h:147:48: error: expression list treated as compound expression in initializer [-fpermissive]
 ink_atomic_increment(pvuint64 mem, Amount value)
                                                ^
../../include/tscore/ink_atomic.h:147:49: error: expected ';' before '{' token
 ink_atomic_increment(pvuint64 mem, Amount value)
                                                 ^
                                                 ;
 {
 ~                                                
../../include/tscore/ink_atomic.h:159:22: error: 'template<class Amount> uint64_t ink_atomic_decrement' conflicts with a previous declaration
 ink_atomic_decrement(pvuint64 mem, Amount value)
                      ^~~~~~~~
../../include/tscore/ink_atomic.h:85:1: note: previous declaration 'Type ink_atomic_decrement(Type*, Amount)'
 ink_atomic_decrement(Type *mem, Amount count)
 ^~~~~~~~~~~~~~~~~~~~
../../include/tscore/ink_atomic.h:159:22: error: 'pvuint64' was not declared in this scope
 ink_atomic_decrement(pvuint64 mem, Amount value)
                      ^~~~~~~~
../../include/tscore/ink_atomic.h:159:22: note: suggested alternative: 'dirent64'
 ink_atomic_decrement(pvuint64 mem, Amount value)
                      ^~~~~~~~
                      dirent64
../../include/tscore/ink_atomic.h:159:43: error: expected primary-expression before 'value'
 ink_atomic_decrement(pvuint64 mem, Amount value)
                                           ^~~~~
../../include/tscore/ink_atomic.h:159:48: error: expression list treated as compound expression in initializer [-fpermissive]
 ink_atomic_decrement(pvuint64 mem, Amount value)
                                                ^
../../include/tscore/ink_atomic.h:159:49: error: expected ';' before '{' token
 ink_atomic_decrement(pvuint64 mem, Amount value)
                                                 ^
                                                 ;
 {
 ~                                                
In file included from P_RecUtils.h:27,
                 from P_RecCore.h:34,
                 from RecCore.cc:29:
../../include/tscore/ink_atomic.h:96:26: error: template-id 'ink_atomic_swap<int64_t>' used as a declarator
 ink_atomic_swap<int64_t>(pvint64 mem, int64_t value)
                          ^~~~~~~
../../include/tscore/ink_atomic.h:96:26: error: 'pvint64' was not declared in this scope
../../include/tscore/ink_atomic.h:96:26: note: suggested alternative: 'printf'
 ink_atomic_swap<int64_t>(pvint64 mem, int64_t value)
                          ^~~~~~~
                          printf
../../include/tscore/ink_atomic.h:96:47: error: expected primary-expression before 'value'
 ink_atomic_swap<int64_t>(pvint64 mem, int64_t value)
                                               ^~~~~
../../include/tscore/ink_atomic.h:96:52: error: expression list treated as compound expression in initializer [-fpermissive]
 ink_atomic_swap<int64_t>(pvint64 mem, int64_t value)
                                                    ^
../../include/tscore/ink_atomic.h:96:1: warning: too many template headers for 'ink_atomic_swap' (should be 0)
 ink_atomic_swap<int64_t>(pvint64 mem, int64_t value)
 ^~~~~~~~~~~~~~~~~~~~~~~~
../../include/tscore/ink_atomic.h:96:53: error: expected ';' before '{' token
 ink_atomic_swap<int64_t>(pvint64 mem, int64_t value)
                                                     ^
                                                     ;
 {
 ~                                                    
../../include/tscore/ink_atomic.h:108:25: error: template-id 'ink_atomic_cas<int64_t>' used as a declarator
 ink_atomic_cas<int64_t>(pvint64 mem, int64_t old, int64_t new_value)
                         ^~~~~~~
../../include/tscore/ink_atomic.h:108:25: error: 'pvint64' was not declared in this scope
../../include/tscore/ink_atomic.h:108:25: note: suggested alternative: 'printf'
 ink_atomic_cas<int64_t>(pvint64 mem, int64_t old, int64_t new_value)
                         ^~~~~~~
                         printf
../../include/tscore/ink_atomic.h:108:46: error: expected primary-expression before 'old'
 ink_atomic_cas<int64_t>(pvint64 mem, int64_t old, int64_t new_value)
                                              ^~~
../../include/tscore/ink_atomic.h:108:59: error: expected primary-expression before 'new_value'
 ink_atomic_cas<int64_t>(pvint64 mem, int64_t old, int64_t new_value)
                                                           ^~~~~~~~~
../../include/tscore/ink_atomic.h:108:68: error: expression list treated as compound expression in initializer [-fpermissive]
 ink_atomic_cas<int64_t>(pvint64 mem, int64_t old, int64_t new_value)
                                                                    ^
../../include/tscore/ink_atomic.h:108:1: warning: too many template headers for 'ink_atomic_cas' (should be 0)
 ink_atomic_cas<int64_t>(pvint64 mem, int64_t old, int64_t new_value)
 ^~~~~~~~~~~~~~~~~~~~~~~
../../include/tscore/ink_atomic.h:108:69: error: expected ';' before '{' token
 ink_atomic_cas<int64_t>(pvint64 mem, int64_t old, int64_t new_value)
                                                                     ^
                                                                     ;
 {
 ~                                                                    
../../include/tscore/ink_atomic.h:123:22: error: 'template<class Amount> int64_t ink_atomic_increment' conflicts with a previous declaration
 ink_atomic_increment(pvint64 mem, Amount value)
                      ^~~~~~~
../../include/tscore/ink_atomic.h:76:1: note: previous declaration 'Type ink_atomic_increment(Type*, Amount)'
 ink_atomic_increment(Type *mem, Amount count)
 ^~~~~~~~~~~~~~~~~~~~
../../include/tscore/ink_atomic.h:123:22: error: 'pvint64' was not declared in this scope
 ink_atomic_increment(pvint64 mem, Amount value)
                      ^~~~~~~
../../include/tscore/ink_atomic.h:123:22: note: suggested alternative: 'printf'
 ink_atomic_increment(pvint64 mem, Amount value)
                      ^~~~~~~
                      printf
../../include/tscore/ink_atomic.h:123:42: error: expected primary-expression before 'value'
 ink_atomic_increment(pvint64 mem, Amount value)
                                          ^~~~~
../../include/tscore/ink_atomic.h:123:47: error: expression list treated as compound expression in initializer [-fpermissive]
 ink_atomic_increment(pvint64 mem, Amount value)
                                               ^
../../include/tscore/ink_atomic.h:123:48: error: expected ';' before '{' token
 ink_atomic_increment(pvint64 mem, Amount value)
                                                ^
                                                ;
 {
 ~                                               
../../include/tscore/ink_atomic.h:135:22: error: 'template<class Amount> int64_t ink_atomic_decrement' conflicts with a previous declaration
 ink_atomic_decrement(pvint64 mem, Amount value)
                      ^~~~~~~
../../include/tscore/ink_atomic.h:85:1: note: previous declaration 'Type ink_atomic_decrement(Type*, Amount)'
 ink_atomic_decrement(Type *mem, Amount count)
 ^~~~~~~~~~~~~~~~~~~~
../../include/tscore/ink_atomic.h:135:22: error: 'pvint64' was not declared in this scope
 ink_atomic_decrement(pvint64 mem, Amount value)
                      ^~~~~~~
../../include/tscore/ink_atomic.h:135:22: note: suggested alternative: 'printf'
 ink_atomic_decrement(pvint64 mem, Amount value)
                      ^~~~~~~
                      printf
../../include/tscore/ink_atomic.h:135:42: error: expected primary-expression before 'value'
 ink_atomic_decrement(pvint64 mem, Amount value)
                                          ^~~~~
../../include/tscore/ink_atomic.h:135:47: error: expression list treated as compound expression in initializer [-fpermissive]
 ink_atomic_decrement(pvint64 mem, Amount value)
                                               ^
../../include/tscore/ink_atomic.h:135:48: error: expected ';' before '{' token
 ink_atomic_decrement(pvint64 mem, Amount value)
                                                ^
                                                ;
 {
 ~                                               
../../include/tscore/ink_atomic.h:147:22: error: 'template<class Amount> uint64_t ink_atomic_increment' conflicts with a previous declaration
 ink_atomic_increment(pvuint64 mem, Amount value)
                      ^~~~~~~~
../../include/tscore/ink_atomic.h:76:1: note: previous declaration 'Type ink_atomic_increment(Type*, Amount)'
 ink_atomic_increment(Type *mem, Amount count)
 ^~~~~~~~~~~~~~~~~~~~
../../include/tscore/ink_atomic.h:147:22: error: 'pvuint64' was not declared in this scope
 ink_atomic_increment(pvuint64 mem, Amount value)
                      ^~~~~~~~
../../include/tscore/ink_atomic.h:147:22: note: suggested alternative: 'uint64_t'
 ink_atomic_increment(pvuint64 mem, Amount value)
                      ^~~~~~~~
                      uint64_t
../../include/tscore/ink_atomic.h:147:43: error: expected primary-expression before 'value'
 ink_atomic_increment(pvuint64 mem, Amount value)
                                           ^~~~~
../../include/tscore/ink_atomic.h:147:48: error: expression list treated as compound expression in initializer [-fpermissive]
 ink_atomic_increment(pvuint64 mem, Amount value)
                                                ^
../../include/tscore/ink_atomic.h:147:49: error: expected ';' before '{' token
 ink_atomic_increment(pvuint64 mem, Amount value)
                                                 ^
                                                 ;
 {
 ~                                                
../../include/tscore/ink_atomic.h:159:22: error: 'template<class Amount> uint64_t ink_atomic_decrement' conflicts with a previous declaration
 ink_atomic_decrement(pvuint64 mem, Amount value)
                      ^~~~~~~~
../../include/tscore/ink_atomic.h:85:1: note: previous declaration 'Type ink_atomic_decrement(Type*, Amount)'
 ink_atomic_decrement(Type *mem, Amount count)
 ^~~~~~~~~~~~~~~~~~~~
../../include/tscore/ink_atomic.h:159:22: error: 'pvuint64' was not declared in this scope
 ink_atomic_decrement(pvuint64 mem, Amount value)
                      ^~~~~~~~
../../include/tscore/ink_atomic.h:159:22: note: suggested alternative: 'uint64_t'
 ink_atomic_decrement(pvuint64 mem, Amount value)
                      ^~~~~~~~
                      uint64_t
../../include/tscore/ink_atomic.h:159:43: error: expected primary-expression before 'value'
 ink_atomic_decrement(pvuint64 mem, Amount value)
                                           ^~~~~
../../include/tscore/ink_atomic.h:159:48: error: expression list treated as compound expression in initializer [-fpermissive]
 ink_atomic_decrement(pvuint64 mem, Amount value)
                                                ^
../../include/tscore/ink_atomic.h:159:49: error: expected ';' before '{' token
 ink_atomic_decrement(pvuint64 mem, Amount value)
                                                 ^
                                                 ;
 {
 ~                                             ```

--- a/include/tscore/ink_atomic.h
+++ b/include/tscore/ink_atomic.h
@@ -93,7 +93,7 @@ extern ink_mutex __global_death;
 
 template <>
 inline int64_t
-ink_atomic_swap<int64_t>(pvint64 mem, int64_t value)
+ink_atomic_swap<int64_t>(int64_t *mem, int64_t value)
 {
   int64_t old;
   ink_mutex_acquire(&__global_death);
@@ -105,7 +105,7 @@ ink_atomic_swap<int64_t>(pvint64 mem, int64_t value)
 
 template <>
 inline bool
-ink_atomic_cas<int64_t>(pvint64 mem, int64_t old, int64_t new_value)
+ink_atomic_cas<int64_t>(int64_t *mem, int64_t old, int64_t new_value)
 {
   int64_t curr;
   ink_mutex_acquire(&__global_death);
@@ -120,7 +120,7 @@ ink_atomic_cas<int64_t>(pvint64 mem, int64_t old, int64_t new_value)
 
 template <typename Amount>
 static inline int64_t
-ink_atomic_increment(pvint64 mem, Amount value)
+ink_atomic_increment(int64_t *mem, Amount value)
 {
   int64_t curr;
   ink_mutex_acquire(&__global_death);
@@ -132,7 +132,7 @@ ink_atomic_increment(pvint64 mem, Amount value)
 
 template <typename Amount>
 static inline int64_t
-ink_atomic_decrement(pvint64 mem, Amount value)
+ink_atomic_decrement(int64_t *mem, Amount value)
 {
   int64_t curr;
   ink_mutex_acquire(&__global_death);
@@ -144,7 +144,7 @@ ink_atomic_decrement(pvint64 mem, Amount value)
 
 template <typename Amount>
 static inline uint64_t
-ink_atomic_increment(pvuint64 mem, Amount value)
+ink_atomic_increment(uint64_t *mem, Amount value)
 {
   uint64_t curr;
   ink_mutex_acquire(&__global_death);
@@ -156,7 +156,7 @@ ink_atomic_increment(pvuint64 mem, Amount value)
 
 template <typename Amount>
 static inline uint64_t
-ink_atomic_decrement(pvuint64 mem, Amount value)
+ink_atomic_decrement(uint64_t *mem, Amount value)
 {
   uint64_t curr;
   ink_mutex_acquire(&__global_death);

--- a/src/traffic_cache_tool/Makefile.inc
+++ b/src/traffic_cache_tool/Makefile.inc
@@ -39,6 +39,7 @@ traffic_cache_tool_traffic_cache_tool_LDADD = \
     $(top_builddir)/src/tscore/.libs/ink_error.o \
     $(top_builddir)/src/tscore/.libs/ink_file.o \
     $(top_builddir)/src/tscore/.libs/ink_memory.o \
+    $(top_builddir)/src/tscore/.libs/ink_mutex.o \
     $(top_builddir)/src/tscore/.libs/ink_string.o \
     $(top_builddir)/src/tscore/.libs/BufferWriterFormat.o \
     $(top_builddir)/src/tscore/.libs/ts_file.o \

